### ISSUE TITLE
Add convenience property for the base unit of a prefixed unit

### ISSUE THREAD

**Issue by embray** [astropy/astropy#4037]
As pointed out in #3835, given a prefixed unit like `u.kpc` there's no easy way (via a public API) to get at the root unit (without the prefix)--i.e. to get `u.pc` from `u.kpc`.

This *can* be gotten from `u.kpc._represents.bases[0]`, but a user-friendly interface for that might be nice to have.

The question is just what to call it, and whether the property should be present on all units or just PrefixUnits?

**Comment by mdboom** [comment]
I would say only on PrefixUnits.  Other units may be composite units and the "first base" is sort of meaningless in that context.

**Comment by vdhelm** [comment]
Hi,

As my first contribution to astropy, I've tried to implement this in the obvious way, by adding the property:

    class PrefixUnit(Unit):
        ...
        @property
        def first_base(self):
            return self._represents.bases[0]

it can in fact also be useful for composite units (go from "km / s" to "m / s") and that is quite easy. In that case it makes sense to define the `first_base` property for `UnitBase` and just return `self` there.

However, we have run into a problem because this does not work for `u.kg` and `u.cm` because these are in fact not instances of `PrefixUnit`.
It's clear from the unit definition code in si.py and cgs.py that this was done deliberately, but it does break this first_base method.
It would be more logical to me if all units where defined without si prefixes (gram and meter) and if you ask for si or cgs units, you just get `PrefixUnit` instances for kg and cm.
How shall we make `first_base` behave correctly for these units?

Cheers,

Edwin

**Comment by algerdnazgul** [comment]
@vdhelm any changes? pull requests?

**Comment by algerdnazgul** [comment]
Maybe close? All in #4037?

**Comment by vdhelm** [comment]
Yes, it would seem that this addition was not exactly what anyone wanted, so I'm good with closing it.

**Comment by pllim** [comment]
Thanks for confirming, @vdhelm .

**Comment by mhvk** [comment]
@pllim - I think you reopened this by mistake, so closing again.

**Comment by pllim** [comment]
@mhvk. Not really. I closed the PR but got some resistance for closing the issue itself, so I reopened it. But if you think this issue is not an issue anymore, then I'm for closing it.

**Comment by mhvk** [comment]
Ah, now I see the whole thread. I actually think the comments in #4110 are somewhat more useful, so linking back to that here.  Specifically, one suggestion I made in there is simply to expose `_represents`:
```
class Unit(UnitBase):
    ...
    @property
    def represents(self):
        return self._represents
```
There seemed to be some agreement that that was fine, and it would solve this somewhat.

**Comment by pllim** [comment]
@mhvk , would you be interested to open a PR with the idea above? Maybe if there is no objections, it might be able to meet 1.2 feature freeze deadline in May!

**Comment by mhvk** [comment]
@pllim - see #4806

**Comment by astrofrog** [comment]
Closed by #4806
